bound/bound.go

```diff
@@ -20,6 +20,26 @@ func (b Bound) IsValid() bool {
 	return b.Min <= b.Max
 }
 
+// Jaccard returns a jaccard index between bounds
+func Jaccard(bounds ...Bound) float64 {
+	if len(bounds) == 0 {
+		return 0.0
+	}
+
+	union := Union(bounds...)
+	if !union.IsValid() || union.Max-union.Min == 0.0 {
```

Comment:
If the union is invalid, return NaN. If the size of the bound is zero, let this fall through. This can be done simply, by writing:
```
u := Union(bounds...)
i : = Intersection(bounds...)
return (i.Max - i.Min) / (u.Max - u.Min)
```

Though we cost a little more in cases where the union is invalid since we don't short circuit the intersection call.